src/mainboard/pcengines/apu1/mainboard.c

```diff
@@ -345,16 +345,17 @@ const char *smbios_mainboard_serial_number(void)
 		return serial;
 
 	dev = pcidev_on_root(4, 0);
-	if (dev)
-		dev = pcidev_path_behind(dev->link_list, PCI_DEVFN(0, 0));
-	if (!dev)
-		return serial;
 
 	/* dev->bus->secondary has 0x100 value, while it should has 0x001.
 	 * This workaround simply corrects it to be valid.
 	 */
 	dev->bus->secondary >>= 8;
```

Comment:
See https://review.coreboot.org/c/coreboot/+/42512

Do we still need to shift the value? Probably not, and this is the root casue of the problem. Please check when the 8bit shift is removed

Reply:
You right. It was the cause of the problem. When I removed it, the serial is calculated correctly.

Reply:
Happens, during rebase we did not remove it.